### ISSUE TITLE
Feature request: associate CAP IDs with clusters and allow querying `clusters` on that basis

### ISSUE THREAD

**Issue by beidelson** [freelawproject/courtlistener#4313]
With the impending deprecation of CAP's APIs, I'm looking at integrating CourtListener as a substitute data source for [Case Viewer](https://www.caseviewer.app/) (cf. #4312). It would help in this transition if the CL clusters that incorporate CAP's data also came with a field for the legacy CAP ID number (and if this field were available for querying the `clusters` endpoint). 

For one thing, Case Viewer stores user history and bookmarks, and these are implemented internally with URLs like `caseviewer://?capID=1905144`. With a `capID` field in the CL data, it would be easy to map these over; without it, the CAP deprecation will break them. Also, having matching ID numbers available from both CAP and CL would make it much easier to test the parsing of the same case from each source during the transition period.

In case it's helpful with this, I noticed that many clusters do already have their CAP IDs embedded in the `filepath_json_harvard` field—e.g., [Palsgraf](https://www.courtlistener.com/api/rest/v4/clusters/3602780/) has this:
```
"filepath_json_harvard": "/storage/harvard_corpus/law.free.cap.ny.248/339.1905144.json"
```
The CAP ID is `1905144`, the last bit of the filename there. Maybe that makes this a very simple transformation, although I don't know if the `filepath_json_harvard` field is uniformly present in the CAP-associated clusters (and in any case it's not possible to query the `clusters` endpoint on this basis as-is).

Thanks for considering.

**Comment by mlissner** [comment]
Thanks Ben. @flooie what are your thoughts on the feasibility of this one? I'm +1 on doing it, but I don't know how hard it would be.

**Comment by flooie** [comment]
Well. The simplest thing would be making the filepath json field usable in the api no?  That seems pretty simple to me.

**Comment by mlissner** [comment]
Yeah, that'd be OK, but kind of clunky. It's not hard just to add the ID and do it right, I'd say.
